crates/sparrow-session/src/session.rs

```diff
@@ -310,13 +329,24 @@ impl Session {
         let mut options = options.to_sparrow_options();
         options.stop_signal_rx = Some(stop_signal_rx);
 
+        let key_hash_inverses = self
+            .key_hash_inverses
+            .get(&group_id)
+            .cloned()
+            .unwrap_or_else(|| {
+                Arc::new(ThreadSafeKeyHashInverse::from_data_type(
+                    primary_grouping_key_type,
+                ))
+            });
+
         // Hacky. Use the existing execution logic. This weird things with downloading checkpoints, etc.
         let progress = rt
             .block_on(sparrow_runtime::execute::execute_new(
                 plan,
                 destination,
                 data_context,
                 options,
+                Some(key_hash_inverses),
```

Comment:
Why are we creating an empty key hash inverse here? Don't we want it to be `None` so we trigger the `load_key_hash_inverse`?

Reply:
Nope -- the loading of the key hash inverse is for old code. Basically, for Python we already have the key has for all the data that has been added to the in-memory session, so we just need to use that one. Any data adding during a materialization is updated (behind the thread-safety of the key hash inverse).